### PR TITLE
Release v0.1.6 - data integration module

### LINKED ISSUE #367: Prepare for scpca-downstream-analyses release v0.1.6
### Preparing for the release

- [x] Are all of the issues planned for this release resolved? If there are any issues that are unresolved, mark this issue as blocked by those on ZenHub.
- [x] Test that the workflow(s) is in good working order and ensure that you have run the latest version of the workflow(s):
  - Ensure that the conda environment can successfully build using `bash setup_envs.sh`.
  - If making additions to the core workflow, test the workflow using the example data with `snakemake --cores 2 --use-conda`
- [x] File a PR from the `development` branch to the `main` branch. This should include all of the changes that will be associated with the next release.

### Creating a release
- [x] On the [releases page](https://github.com/AlexsLemonade/scpca-downstream-analyses/releases), choose `Draft a new release`.
- [x] In `Choose a tag`, type a new release number using semantic versioning (vX.X.X) (you did update the title of this issue to match, right?), then click `Create a new tag: vX.X.X on publish`.
- [ ] Write a description of the major changes in this release. You may want to start with the auto-generated release notes to save time.
- [x] **Optional**: If not all issues have been addressed, save a draft to return to later.
- [ ] Publish the release!

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #367 

**What is the purpose of these changes?**

This PR adds the data integration module changes made on the `development` branch to the `main` branch.

**What changes did you make?**

The following changes are included here:
- the addition of `optional-integration-analysis/merge-sce.R`
- the addition of `optional-integration-analysis/perform-integration.R`
- the addition of the template `optional-integration-analysis/integration-report-template.Rmd`
- the addition of the `integration.snakefile`, along with an `integration_config.yaml` file
- a README added to the integration module and relevant doc updates in `additional-parameters.md` and the main README
- modifications to make `use-conda` default and relevant doc updates throughout the repo
- modifications to the `utils/sync-results.sh` script to account for syncing the integration example results files to s3
- relevant package and dependency updates

**Any comments, concerns, or questions important for reviewers**
This one was pretty straightforward but let me know if it looks like I am missing anything!

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [x] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: optional-integration-analysis/merge-sce.R [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+# Check that integration group is provided
+if(is.null(opt$integration_group)) {
+  stop("The `integration_group` is missing and required for identifying which objects should be merged together.")
+} else if (!opt$integration_group %in% input_metadata$integration_group) {
+  stop("The provided `integration_group` is missing from the input metadata file.")
+} else {
+  # Filter the metadata to include only information relevant to the specified integration group
+  input_metadata <- input_metadata %>%
```
Comment (allyhawkins) [inline optional-integration-analysis/merge-sce.R:88]: Just happened to notice this is a lingering magrittr pipe, but we dont load magrittr here.

File: optional-integration-analysis/perform-integration.R [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+# Read in merged SCE -----------------------------------------------------------
+
+merged_sce <- readr::read_rds(opt$merged_sce_file)
+
+# Integrate SCEs ---------------------------------------------------------------
+
+# Split up string of integration methods
+integration_methods <-  stringr::str_split(opt$integration_method, ",") %>%
```
Comment (allyhawkins) [inline optional-integration-analysis/perform-integration.R:94]: This script also uses the magrittr pipe. We should be consistent within this module and use the base R pipe.

**Comment by allyhawkins** [comment]
This should now have all the integration updates, conda changes, and new version of `scpcaTools`.

**Review by jashapiro — APPROVED** [review]
Lgtm

**Pull request merged**